app/webpacker/components/RegistrationsV2/Register/RegistrationRequirements.jsx

```diff
@@ -0,0 +1,102 @@
+import { UiIcon } from '@thewca/wca-components';
+import { marked } from 'marked';
+import React, {
+  useContext, useEffect, useMemo, useState,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  Accordion,
+  Button,
+  Form,
+  Message,
+  Segment,
+  Transition,
+} from 'semantic-ui-react';
+import { CompetitionContext } from '../Context/competition_context';
+
+export default function RegistrationRequirements({ nextStep }) {
+  const { competitionInfo } = useContext(CompetitionContext);
+
+  const [generalInfoAcknowledged, setGeneralInfoAcknowledged] = useState(false);
+  const [regRequirementsAcknowledged, setRegRequirementsAcknowledged] = useState(false);
+
+  const [showRegRequirements, setShowRegRequirements] = useState(false);
+
+  const { t } = useTranslation();
+
+  const setFromCheckbox = (data, setState) => {
+    const { checked } = data;
+    setState(checked);
+  };
+
+  const handleAccordionClick = () => {
+    setShowRegRequirements((oldShowRegRequirements) => !oldShowRegRequirements);
+  };
+
+  const buttonDisabled = useMemo(() => (
+    !generalInfoAcknowledged
+      || (competitionInfo.extra_registration_requirements
+        && !regRequirementsAcknowledged)
+  ), [
+    competitionInfo.extra_registration_requirements,
+    generalInfoAcknowledged,
+    regRequirementsAcknowledged,
+  ]);
+
+  useEffect(() => {
+    if (generalInfoAcknowledged) {
+      setShowRegRequirements(true);
+    }
+  }, [generalInfoAcknowledged, setShowRegRequirements]);
+
+  return (
+    <Segment basic>
+      <Form onSubmit={nextStep}>
+        <Form.Checkbox
+          checked={generalInfoAcknowledged}
+          onClick={(_, data) => setFromCheckbox(data, setGeneralInfoAcknowledged)}
```

Comment:
If you use the custom hook instead of `useState` hook (see comment above) you can directly pass in the setter and the hook logic will figure everything out by itself.